content/departments/engineering/dev/tools/grpc_cs_enablement.md

```diff
@@ -0,0 +1,165 @@
+# gRPC CS Enablement Guide / Primer
+
+## What is gRPC?
+
+[gRPC](https://grpc.io/) is a high-performance [Remote Procedure Call (RPC)](https://en.wikipedia.org/wiki/Remote_procedure_call) framework developed by Google.
+
+### Sidebar: What is an RPC (Remote Procedure Call)?
+
+At its most basic level, an [RPC](https://en.wikipedia.org/wiki/Remote_procedure_call) allows one program to cause a function to execute in another program (commonly on another physical machine).
+
+In simpler terms, it's a way for different parts of a system, which could be located on different machines, to talk to each other and perform tasks.
+
+Imagine being able to “press a button in one” application to get information or perform an action in another application located somewhere else; that's the essence of an [RPC](https://en.wikipedia.org/wiki/Remote_procedure_call).
+
+Here is a pseudo-code example of what an [RPC](https://en.wikipedia.org/wiki/Remote_procedure_call) can look like:
+
+```javascript
+// On Server Machine
+function add(a, b) {
+  return a + b
+}
+
+/// ---------------
+
+// On Client Machine
+result = remoteCall('add', 5, 3) // Makes an RPC to the 'add' function on the server
+print(result) // Output will be 8
+```
+
+## What is changing?
+
+Starting in Sourcegraph `5.2.X` , we are switching to use gRPC for Sourcegraph’s **_internal communication (service to service only)_** by default.
+
+See the table below for more of the rollout details:
+
+### Rollout timeline
+
+| Sourcegraph Version                   | gRPC Feature Status                                                                   |
+| ------------------------------------- | ------------------------------------------------------------------------------------- |
+| 5.2.X (releasing October 4th, 2023)   | On by default, can be disabled via feature flag (see “enabling/disabling GRPC below”) |
+| 5.3.X (releasing December 14th, 2023) | On by default, cannot be turned off                                                   |
+
+## Why are we making this change?
+
+As opposed to [REST](https://en.wikipedia.org/wiki/Representational_state_transfer) (the pattern we currently use for [RPCs](https://en.wikipedia.org/wiki/Remote_procedure_call)), gRPC offers the following benefits and more:
```

Comment:
Maybe worth mentioning in `()` that we don't even use REST consistently... Or that we don't have any consistent APIs across all services ;)